python-modules/cis_publisher/cis_publisher/auth0.py

```diff
@@ -233,12 +233,10 @@ def fetch_az_users(self, user_ids=None):
             az_query = az_query + t + 'identities.connection:"{}"'.format(azc)
             t = " OR "
         az_query = az_query + ")"
-        az_query = az_query + " AND NOT ("
-        t = ""
-        for azc in self.az_blacklisted_connections:
-            az_query = az_query + t + 'identities.connection:"{}"'.format(azc)
-            t = " OR "
-        az_query = az_query + ")"
+        # NOTE XXX: There is no way to tell auth0's ES "don't include matches where the first identity.connection is a
+        # blacklisted connection", so we do this instead. This 100% relies on auth0 user_ids NOT being opaque,
```

Comment:
Is there any documentation that can be linked to that guarantees that these IDs _aren't_ opaque for now?  It would be useful to reference here (i.e. link to directly in this comment) in case a bug appears here in the future.

Reply:
No :( From a data point of view they should be considered opaque
From a pragmatic reality they won't be with auth0. I would hope that if they change it we have another way to identify these accounts - otherwise, the code would simply "do more work than it needs", i.e., it will match wrongly, convert the auth0 users into our standardized user profiles, then a function checks again if it's importing the wrong users and drop them (that function is already there as a safety)